source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+---
+title: route2
+containerClass: experimental
+---
+
+{% note warning %}
+{% fa fa-warning orange %} **This is an experimental feature. In order to use it, you must manually configure the `experimentalNetworkMocking` option to `true`.** See {% issue 687 %} for more details.
+{% endnote %}
+
+Use `cy.route2()` to manage the behavior of network requests at the network layer.
+
+## Comparison to cy.route()
+
+Unlike {% url `cy.route()` route %}, `cy.route2()`:
+
+- can mock all types of network requests (i.e., Fetch API, page loads, XMLHttpRequests, etc.)
+- no longer requires {% url `cy.server()` server %} before use
+
+# Syntax
+
+```javascript
+cy.route2(requestMatcher)
+cy.route2(requestMatcher, response)
+cy.route2(method, requestMatcher)
+cy.route2(method, requestMatcher, response)
+```
+
+## Usage
+
+**{% fa fa-check-circle green %} Correct Usage**
+
+```javascript
+cy.route2('/users/**')
+```
+
+## Arguments
+
+**{% fa fa-angle-right %} requestMatcher** ***(String, Glob, RegExp, Object)***
+
+The `requestMatcher` argument allows you to listen for a route matching a specific URL or pattern.
+
+For simple route matching, passing a string, glob, or RegExp is the simplest method:
+
+```js
+// String
+cy.route2('/users)
+
+// Regex
+cy.route2(/users\/\d+/)
+
+// Glob to match all paths in the /user path
+cy.route2('/users/**')
+
+// https://localhost:8080/users/1b2c3            <-- matches
+// https://localhost:8080/users/profile/edit  <-- matches
+// https://localhost:8080/users/transaction?month=03&year=2020 <-- matches
+
+```
+
+However, if you need to match additional properties on the route, you can pass an object instead.
+
+```js
+cy.route2({
+  path: '/users/**',
+  method: 'POST',
+})
+```
+
+The following contains a complete list of available properties you can verify against:
+
+Option | Default | Description
+--- | --- | ---
+`auth` | `null` | Object with a `username` and `password` to match HTTP basic authentication
+`headers` | `null` | Object to match client request headers
+`hostname` | `null` | String, RegExp, or Glob to match based on requested hostname
+`method` | `'ALL'` | Match based on HTTP method (`GET`, `POST`, `PUT`, etc.)
+`path` | `null` | String, RegExp, or Glob to match on request path after the hostname, including query params
+`pathname` | `null` | String, RegExp, or Glob to match on request path after the hostname, without query params
+`port` | `null` | Match based on requested port number
+`query` | `null` | Object to match based on query parameters
+`url` | `null` | String, RegExp, or Glob to match based on full request URL
+
```

Comment:
Probably worth mentioning that globbing uses the minimatch API, so that users can understand how globbing works: https://github.com/isaacs/minimatch

https://github.com/cypress-io/cypress/blob/50a45c1b707ef0ccb0447b94ecb04946e994a4db/packages/net-stubbing/lib/server/intercept-request.ts#L63-L63

Reply:
still need a mention here - passing a string to `auth.username`, `auth.password`, `headers.*`, `hostname`, `method`, `path`, `pathname`, `url` (any string property matcher) will use minimatch to match under the hood

Reply:
here is the backend's implementation for matching `StringMatcher` fields: https://github.com/cypress-io/cypress/blob/0d60f7cd3ede4c5a79e151b646fa2377a7ddb16c/packages/net-stubbing/lib/server/intercept-request.ts#L41-L64